File: app/src/main/java/com/nptel/courses/online/diffutils/VideoDiffCallback.java
package com.nptel.courses.online.diffutils;

import com.nptel.courses.online.entities.Video;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

public class VideoDiffCallback extends DiffUtil.ItemCallback<Video> {

    @Override
    public boolean areItemsTheSame(@NonNull Video oldItem, @NonNull Video newItem) {
        return newItem.getYoutubeVideoId().equals(oldItem.getYoutubeVideoId());
    }

    @Override
    public boolean areContentsTheSame(@NonNull Video oldItem, @NonNull Video newItem) {
        return newItem.getTitle().equals(oldItem.getTitle()) && newItem.getDescription().equals(oldItem.getDescription());
    }
}

File: app/src/main/java/com/nptel/courses/online/diffutils/VideoDiffCallback.kt
package com.nptel.courses.online.diffutils

import androidx.recyclerview.widget.DiffUtil
import com.nptel.courses.online.entities.Video

class VideoDiffCallback : DiffUtil.ItemCallback<Video>() {
    override fun areItemsTheSame(oldItem: Video, newItem: Video): Boolean {
        return newItem.youtubeVideoId == oldItem.youtubeVideoId
    }

    override fun areContentsTheSame(oldItem: Video, newItem: Video): Boolean {
        return newItem.title == oldItem.title && newItem.description == oldItem.description
    }
}